projects/igniteui-angular/src/lib/select/select.component.ts

```diff
@@ -312,9 +377,51 @@ export class IgxSelectComponent extends IgxDropDownComponent implements IgxSelec
 
     /** @hidden @internal */
     public onBlur(): void {
+        if (this.ngControl && !this.ngControl.valid) {
+            this.valid = IgxSelectState.INVALID;
+        } else {
+            this.valid = IgxSelectState.INITIAL;
+        }
+
         if (!this.collapsed) {
             this.toggleDirective.close();
         }
     }
+
+    protected onStatusChanged() {
+        if ((this.ngControl.control.touched || this.ngControl.control.dirty) &&
+            (this.ngControl.control.validator || this.ngControl.control.asyncValidator)) {
+            this.valid = this.ngControl.valid ? IgxSelectState.VALID : IgxSelectState.INVALID;
+        }
+    }
+    /**
+     * @hidden @internal
+     */
+    public ngOnInit() {
+        try {
+            this.ngControl = this._injector.get(NgControl);
+        } catch (e) { }
+    }
+
+    /**
+     * @hidden @internal
+     */
+    public ngAfterViewInit() {
+        if (this.ngControl) {
+            this._statusChanges$ = this.ngControl.statusChanges.subscribe(this.onStatusChanged.bind(this));
+        }
+    }
+
+    /**
+     * @hidden @internal
+     */
+    public ngOnDestroy() {
+        this.destroy$.complete();
```

Comment:
`this.destroy$` isn't needed - it's only initialized with the component and then completed in `onDestroy`